src/main/java/de/rwth/idsg/steve/config/ApiAuthenticationManager.java

```diff
@@ -99,8 +113,19 @@ public void commence(HttpServletRequest request,
         response.getWriter().print(jacksonObjectMapper.writeValueAsString(apiResponse));
     }
 
+    private UserDetails getFromCacheOrDatabase(String username) {
```

Comment:
Why do you set the cache logic here instead of `webUserService.loadUserByUsernameForApi`?

I don't know if it is a choice to not use it but Spring has its own way of doing caches: https://spring.io/guides/gs/caching 
And you can still use guava under to wood if you want it: https://docs.spring.io/spring-framework/docs/4.2.x/javadoc-api/org/springframework/cache/guava/GuavaCacheManager.html

Reply:
@goekay Just for my technical knowledge because I didn't use them before: why do you use guava directly and not hidden behind spring cache?

Reply:
Thanks for the answer! 👍 

> why introduce spring magic?

That could be true for almost every spring import ;)